### PR TITLE
Rename chdir to package-path

### PR DESCRIPTION
**Pull request by hartbit**

### CODE REVIEW

**Comment by hartbit** [comment]
While implementing this, I've had doubts about the name. It can be ambiguously understood to mean the path to the manifest file. Ideas: `--package-root`, `--package-dir`.

**Comment by ddunbar** [comment]
We discussed `--package-dir{,ectory}` as well. I personally find --package-directory pretty longwinded but maybe that is ok. @rballard do you have an opinion here?

I agree with the sentiment, but I'm not sure it will end up being much of a problem in practice.

File: Sources/Commands/SwiftTool.swift [hunk 1]
```diff
@@ -232,6 +238,10 @@ public class SwiftTool<Options: ToolOptions> {
         self.buildPath = getEnvBuildPath() ??
             customBuildPath ??
             (packageRoot ?? currentWorkingDirectory).appending(component: ".build")
+        
+        if options.chdir != nil {
+            warning(message: "-C/--chdir is deprecated; use --package-path instead")
```
Comment (aciidb0mb3r) [inline Sources/Commands/SwiftTool.swift]: Not related to this PR but we should think about adding deprecated support to the parser itself. That might be handy.

File: Sources/Commands/SwiftTool.swift [hunk 2]
```diff
@@ -150,8 +150,14 @@ public class SwiftTool<Options: ToolOptions> {
         binder.bind(
             option: parser.add(
                 option: "--chdir", shortName: "-C", kind: PathArgument.self,
-                usage: "Change working directory before any other operation"),
+                usage: "Deprecated, use --package-path instead"),
```
Comment (aciidb0mb3r) [inline Sources/Commands/SwiftTool.swift]: We should hide this option completely. Set usage to `nil` to hide it.

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Comment by hartbit** [comment]
@aciidb0mb3r I removed the usage and I changed from calling `warning()` to using the diagnostics engine

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Pull request merged**